core/amber/src/main/scala/edu/uci/ics/texera/workflow/common/storage/ScanSourceFileResolver.scala

```diff
@@ -0,0 +1,37 @@
+package edu.uci.ics.texera.workflow.common.storage
+
+import java.nio.file.{Files, Path, Paths}
+import edu.uci.ics.amber.engine.common.storage.DatasetFileDocument
+import scala.util.{Failure, Success, Try}
+
+object ScanSourceFileResolver {
+
+  case class SourceFileNotFound(message: String) extends Exception(message)
+
+  /**
+    * Attempts to resolve the given fileName to either a local file path or a DatasetFileDocument.
+    *
+    * @param fileName the name of the file to resolve
+    * @return a tuple of (Option[Path], Option[DatasetFileDocument])
+    */
+  def resolve(fileName: String): (Option[String], Option[DatasetFileDocument]) = {
+    val filePath = Paths.get(fileName)
+
+    // Check if the file exists in the local filesystem
+    if (Files.exists(filePath)) {
```

Comment:
This is redundant. Just attempt to open it is enough? You can catch the error and attempt using the second method. 

Resolvers=[localResolveFunc, remoteResolveFunc]

Output:Either[String, Document]

For resolver in resolvers:
    Try{
     Output=resolver(inputPath)
     } match {
  Success => return output
  Fail => surpass the error
   }
Throw an error

Reply:
changed